zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java

```diff
@@ -206,9 +209,9 @@ public Factory(WebClient httpClient) {
     try (SafeCloseable ignored =
            Clients.withContextCustomizer(ctx -> ctx.logBuilder().name(name))) {
       HttpRequestWriter httpRequest = HttpRequest.streaming(request.headers());
-      response = httpClient.execute(httpRequest);
       request.writeBody(httpRequest::tryWrite);
       httpRequest.close();
+      response = httpClient.execute(httpRequest);
```

Comment:
Is this change needed? I think the previous pattern was to allow streaming the request to Elasticsearch while serializing spans. Now, this is effectively `AggregatedHttpRequest` from what I understand, which we could switch to instead, but it would be better if we can use the streaming pattern here this may be a significant performance regression.

Reply:
Of course stability is more important - just confirming whether this was actually required though.

Reply:
Yeah, the problem was that the request is aborted if the client receives the response before it the request is fully sent. How about using `whenConsumed()` then?

Reply:
But didn't you change it so the test doesn't do that anymore? Is there a problem in the business logic here with the previous pattern, I thought it's the idiomatic way of streaming a request in Armeria isn't it?

Reply:
>  I thought it's the idiomatic way of streaming a request in Armeria isn't it?

For the streaming request, yeah I think so. It's just that the request is aborted while streaming if the client receives the corresponding response.

Previously, we call the `writeBody` after sending the request headers which means that the streaming body might be sent one by one.
But with this change right now, yeah, we buffer all the stream body into the queue first and then send them all.

If we can use `whenConsumed()` we can accomplish the same way in which the streaming body is sent one by one exactly.

Reply:
@anuraaga Changed it. PTAL.